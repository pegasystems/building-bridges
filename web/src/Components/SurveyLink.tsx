import React from 'react';

interface SurveyLinkProps {
    link: string
}

interface SurveyLinkState {
    textToShow: string
}

export default class SurveyLink extends React.Component<SurveyLinkProps, SurveyLinkState> {
    state = {
        textToShow: 'Copy link',
    }

    copyToClipboard = () => {
        const textField = document.createElement('textarea');
        textField.innerText = this.props.link;
        document.body.appendChild(textField);
        textField.select();
        document.execCommand('copy');
        textField.remove();
        this.setState({textToShow: 'Copied!'});
        setTimeout(() => {
            this.setState({textToShow: 'Copy link'});
        }, 2000);
    };

    render(): JSX.Element {
        return (<i className="copy" onClick={this.copyToClipboard}>{this.state.textToShow}</i>);
    }
}
