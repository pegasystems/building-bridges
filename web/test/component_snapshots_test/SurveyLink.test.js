// Link.react.test.js
import React from 'react';
import SurveyLink from '../../src/Components/SurveyLink';
import renderer from 'react-test-renderer';

document.execCommand = jest.fn()

test('Survey link renders properly test', () => {
  const component = renderer.create(
    <SurveyLink link="https://www.example.com"/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Survey link shows info about copied link', () => {
  const component = renderer.create(
      <SurveyLink link="https://www.example.com"/>
  );
  let tree = component.toJSON();

  // manually trigger the callback
  tree.props.onClick();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});