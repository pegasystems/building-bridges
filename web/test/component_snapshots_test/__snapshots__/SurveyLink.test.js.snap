// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Survey link renders properly test 1`] = `
<i
  className="copy"
  onClick={[Function]}
>
  Copy link
</i>
`;

exports[`Survey link shows info about copied link 1`] = `
<i
  className="copy"
  onClick={[Function]}
>
  Copied!
</i>
`;
